addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java

```diff
@@ -157,6 +157,7 @@ protected final boolean isEquals(@Nullable Configuration configuration, String a
 
     public final @Nullable OutboundSpec getCommandSpec(Configuration configuration, KNXTypeMapper typeHelper,
             Type command) throws KNXFormatException {
+        logger.trace("getCommandSpec testing Keys '{}' for command '{}'", getAllGAKeys(), command);
```

Comment:
@lewie Just a (important?) question how heavy is the `getAllGAKeys()` method, it is called an extra time even when the logging level is non-trace is this okay in this case?

Reply:
getAllGAKeys means normally one or maximum 5 Keys. For logger.trace it is helpful at this position.

Reply:
Yes but does it do a calculation or is it a more less static call?

Reply:
It is static:
```
    @Override
    protected Set<String> getAllGAKeys() {
        return Stream.of(SWITCH_GA, POSITION_GA, INCREASE_DECREASE_GA, HSB_GA).collect(toSet());
    }
```

Reply:
So my point is indeed that it now will always construct that set twice (also when running with info logging) so maybe a next time you could fix the code that it does collect the set only once.

Reply:
Yes, this makes sense.